packages/grid/_modules_/grid/hooks/features/columns/useGridColumns.ts

```diff
@@ -165,16 +168,25 @@ export function useGridColumns(columns: GridColumns, apiRef: GridApiRef): void {
     [forceUpdate, getColumnFromField, updateColumns],
   );
 
-  const moveColumn = React.useCallback(
+  const setColumnIndex = React.useCallback(
     (field: string, targetIndexPosition: number) => {
       logger.debug(`Moving column ${field} to index ${targetIndexPosition}`);
       const oldIndexPosition = gridState.columns.all.findIndex((col) => col === field);
```

Comment:
What if we call this function with the same index?
it should return directly and not fire any event